### PR TITLE
feat: create department

### PR DESCRIPTION
**Pull request by StarryFolf**

### CODE REVIEW

**Review by Vzart — APPROVED** [review]
Nothing seems out of order, good job

File: src/Application/Departments/Commands/CreateDepartment/CreateDepartmentCommand.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        _uow = uow;
+        _mapper = mapper;
+    }
+
+    public async Task<DepartmentDto> Handle(CreateDepartmentCommand request, CancellationToken cancellationToken)
+    {
+        var entity = new Department
+        {
+            Id = Guid.NewGuid(),
```
Comment (ChienNQuang) [inline src/Application/Departments/Commands/CreateDepartment/CreateDepartmentCommand.cs:29]: should separate id generation and check

**Review by ChienNQuang — CHANGES_REQUESTED** [review]
Need a little change

**Review by ChienNQuang — APPROVED** [review]
Good job

File: src/Application/Common/Interfaces/Repositories/IDepartmentRepository.cs [hunk 2]
```diff
@@ -4,5 +4,6 @@ namespace Application.Common.Interfaces.Repositories;
 
 public interface IDepartmentRepository
 {
+    Task<Department> CreateDepartmentAsync(Department department);
     Task<Department> GetByIdAsync(Guid id);
```
Comment (kaitoz11) [inline src/Application/Common/Interfaces/Repositories/IDepartmentRepository.cs:8]: Get what by id?

**Review by kaitoz11 — APPROVED** [review]

**Pull request merged**